app/uk/gov/hmrc/economiccrimelevyregistration/navigation/RelevantAp12MonthsPageNavigator.scala

```diff
@@ -26,13 +26,27 @@ class RelevantAp12MonthsPageNavigator extends PageNavigator {
     navigate(eclRegistrationModel.registration, NormalMode)
 
   override protected def navigateInCheckMode(eclRegistrationModel: EclRegistrationModel): Call =
-    navigate(eclRegistrationModel.registration, CheckMode)
+    navigate(eclRegistrationModel.registration, CheckMode, eclRegistrationModel.hasRegistrationChanged)
 
-  private def navigate(registration: Registration, mode: Mode): Call =
-    registration.relevantAp12Months match {
-      case Some(true)  => routes.UkRevenueController.onPageLoad(mode)
-      case Some(false) => routes.RelevantApLengthController.onPageLoad(mode)
-      case _           => routes.NotableErrorController.answersAreInvalid()
+  private def navigate(registration: Registration, mode: Mode, hasRegistrationChanged: Boolean = true): Call =
```

Comment:
Why do you need to default this? The model already has a default.

Reply:
So I can omit it in `NormalMode`. Ut will always be `true` in that case.

Reply:
It'd be easier to understand if it wasn't defaulted. Passing it as true would make it clear what is happening in normal mode, instead of just magic going on from defaulted values.